### PR TITLE
REF: DataFrame.isna internals access

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4478,19 +4478,20 @@ def _maybe_casted_values(index, labels=None):
 
     @Appender(_shared_docs["isna"] % _shared_doc_kwargs)
     def isna(self) -> "DataFrame":
-        return super().isna()
+        result = self._constructor(self._data.isna(func=isna))
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: This should be isna or isna_old depending on the settings?

Comment (jorisvandenbossche) [reply pandas/core/frame.py]: (and since that is handled in the `isna` function, I am not fully sure it will be easier to do it here)

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py]: > This should be isna or isna_old depending on the settings?

No, the _isna/_isna_old is reflected in `isna`

Comment (jorisvandenbossche) [reply pandas/core/frame.py]: Ah, I missed that. So it's basically calling that recursively on its parts. OK!

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**